src/server/req_manager.c

```diff
@@ -810,64 +813,73 @@ mgr_set_attr2(attribute *pattr, attribute_def *pdef, int limit, svrattrl *plist,
 	if (plist == (struct svrattrl *)0)
 		return (PBSE_NONE);
 
+	/*
+	 * We have multiple attribute lists in play here.  pre_copy is used
+	 * to copy the attributes into pattr prior to calling the action functions.
+	 * This means the object passed to the action function is fully up to date.
+	 * new is attribute list which the action functions are called with.  new
+	 * may be modified by the action functions, so we need to copy these to pattr
+	 * a second time.  Lastly we have attr_save.  It is a copy of the object's
+	 * attributes prior to any change.  If any action function fails, we copy
+	 * attr_copy back to the real attributes before leaving the function.
+	 */
+
 	new = (attribute *)calloc((unsigned int)limit, sizeof(attribute));
 	if (new == (attribute *)0)
 		return (PBSE_SYSTEM);
-	/*
-	 * decode the new attribute values which are in the request,
-	 * copy the corresponding current attribute into a holding array
-	 * and update it with the newly decoded value
-	 */
 
-	/* Below says if 'allow_unkresc' is TRUE, then set 'unkn' param */
-	/* of attr_atomic_set() to '1'; otherwise, set it to '-1' meaning */
-	/* not to allow unknown resource. */
-	if ((rc =attr_atomic_set(plist, pattr, new, pdef, limit, (allow_unkresc?1:-1), privil, bad)) != 0) {
+	/* Below says if 'allow_unkresc' is TRUE, then set 'unkn' param
+	 * of attr_atomic_set() to '1'; otherwise, set it to '-1' meaning
+	 * not to allow unknown resource
+	 */
+	if ((rc = attr_atomic_set(plist, pattr, new, pdef, limit, (allow_unkresc?1:-1), privil, bad)) != 0) {
```

Comment:
There seems to be a problem here, if we allow unknown resources then we should pass the last index value where we want to append the resource. '1' will probably make attr_atomic_set use wrong resource index (probably 'mem' resource) to decode/set the resource.